conference/react-js/src/components/Header/index.jsx

```diff
@@ -1,31 +1,64 @@
 import React, { PureComponent } from 'react';
+import classNames from 'classnames';
 import SignInForm from '../../containers/SignInForm';
 
 class Header extends PureComponent {
   constructor() {
     super();
-
-    this.onButtonAccountClick = this.onButtonAccountClick.bind(this);
-    this.state = { visible: true };
+    this.state = {
+      visible: false,
+    };
   }
 
-  onButtonAccountClick() {
-    const currentState = this.state.visible;
-    this.setState({ visible: !currentState });
-  }
+  onButtonAccountClick = () => {
+    this.setState(
+      { visible: !this.state.visible },
+      () => {
+        document.removeEventListener('click', this.closeSignIn);
+        if (this.state.visible) {
+          document.addEventListener('click', this.closeSignIn);
+        }
+      },
+    );
+  };
+
+  closeSignIn = (event) => {
+    const formContainer = document.querySelector('.menu-container__content');
```

Comment:
This is not react way to write code.
You must forget about listeners, selectors etc.
We can contact and discuss this functionality